src/ngraph/specialize_function.hpp

```diff
@@ -107,5 +107,6 @@ namespace ngraph
         specialize_function(std::shared_ptr<Function> f,
                             const std::vector<element::Type>& parameter_element_types,
                             const std::vector<PartialShape>& parameter_shapes,
-                            const std::vector<void*>& parameter_values);
+                            const std::vector<void*>& parameter_values,
+                            bool constant_folding = true);
```

Comment:
Can we make the default 'false'. ConstantFolding can be quite slow since we use reference kernels and it might come as a surprise to users if we enable by default.